DS-Session1-Question2/src/utility/LinkedList.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * @author Abhishek Bishnoi
+ * @slowScale Jul 29, 2019
+ * This Class Provides Dynamic List as where,
+ * it provides various methods like add, toString etc etc
+ * This class Main purpose is to check a loop in Linked List
+ */
+package utility;
+
+public class LinkedList {
+	Node headNode=null;
+	Node lastNode=null;
+	/**
+	 * This Function is used to Add Data into Node And Connecting them with Next Node 
+	 * @param node : Node to adding in List
+	 * @return boolean: true or false for operation success or not
+	 * @exception : when adding of element can't be done
+	 */
+	public boolean add(Node node){
+		if(node==null){
+			throw new AssertionError("Node Can't Be Null");
+		}
+		try{
+			if(lastNode==null){
+				headNode=node;
+				lastNode=node;
+			}else{
+				lastNode.next = node;
+				lastNode = node;
+			}
+			return true;
+		}catch(Exception exception){
+			throw new AssertionError("Error in Insertion");
+		}
+	}
+
+	/**
+	 * This Function is use to calculate if their is any loop in the Linked List or Not 
+	 * @return
+	 */
+	public boolean isLoop(){
+		Node currentNode = headNode;
+		Node checkNode = null;
+		int slowScale = 0; // tracing flag 
+		int fastScale = 2; // fast tracing flag
+		// these flags are use to trace nodes  
+		do {
+		   if (checkNode == currentNode) return true; 
```

Comment:
How two nodes are equal ? == will check the reference, but we should check the value of the nodes. If value of the two nodes are equal, then they should be equal.

Reply:
Sir this problem is about loop in a linkec list and loop only can possible if any node's reference occured twice in a program.